safir/src/safir/uws/_service.py

```diff
@@ -271,20 +274,110 @@ async def get(
 
         return job
 
+    async def get_error(self, user: str, job_id: str) -> UWSJobError | None:
+        """Get the error for a job, if any.
+
+        Parameters
+        ----------
+        user
+            User on behalf this operation is performed.
+        job_id
+            Identifier of the job.
+
+        Returns
+        -------
+        UWSJobError or None
+            Error information for the job, or `None` if the job didn't fail.
+
+        Raises
+        ------
+        PermissionDeniedError
+            If the job ID doesn't exist or is for a user other than the
+            provided user.
+        """
+        job = await self._storage.get(job_id)
+        if job.owner != user:
+            raise PermissionDeniedError(f"Access to job {job_id} denied")
+        return job.error
+
+    async def get_summary(
+        self,
+        user: str,
+        job_id: str,
+        *,
+        signer: ResultStore | None = None,
+        wait_seconds: int | None = None,
+        wait_phase: ExecutionPhase | None = None,
+    ) -> JobSummary:
+        """Retrieve a job's XML model.
+
+        This also supports long-polling, to implement UWS 1.1 blocking
+        behavior, and signing of results.
+
+        Parameters
+        ----------
+        user
+            User on behalf this operation is performed.
+        job_id
+            Identifier of the job.
+        signer
+            If provided, generate signed URLs for the results by using the
+            given signing object. If no object is supplied, no URLs will be
+            included in the result list.
+        wait_seconds
+            If given, wait up to this many seconds for the status to change
```

Comment:
I'm hoping to get rid of this retry loop entirely by doing something a bit more sophisticated, since we should at least block on completion of the arq job before we start waiting for the database update to happen. If I end up still needing a retry loop in Wobbly after that, I'll take a look. Thanks!